apps/build.groovy

```diff
@@ -29,15 +29,7 @@ def build(boolean verboseMaven) {
 	dir ('apps') {
 		quietFlags = verboseMaven ? '' : '--quiet --batch-mode'
 
-		sh "mvn ${quietFlags} --threads 1C --update-snapshots -Dmaven.test.failure.ignore clean verify"
-		/*
-		 * Fingerprint the output artifacts and archive the test results.
-		 *
-		 * Archiving the artifacts here would waste space, as the build deploys them to the local Maven repository.
-		 */
-		fingerprint '**/target/*.jar,**/target/*.war,**/target/*.zip'
-		junit testResults: '**/target/*-reports/TEST-*.xml', keepLongStdio: true
```

Comment:
Is there any type of configuration outside of github which could break tests (and the environment) which we would lose visibility of by removing this step? (Alternatively, could we just run tests in _one_ environment during deploys instead?) I can't immediately think of too many cases of non-code config breaking the tests, outside of maybe database or aws config changes between last merged commit and deploy. Keeping our merge > deploy cycle short will minimize these risks, but still something to think about.

Reply:
Thanks for the feedback!

> Is there any type of configuration outside of github which could break tests (and the environment) which we would lose visibility of by removing this step?

In some earlier discussions about eliminating the tests as part of the Jenkins pipeline, there was some discussion of whether it has ever been the case that we've had a failure in Jenkins but the same change set passed in GitHub Actions. Ignoring intermittent failures in flakey ITs, the _collective_ recollection suggested that this happened around the time of upgrading Java/JDK versions...

Further, others said that ensuring that testing with two different CI mechanisms is not a common requirement and has really just been accidental in this case.

Also of note, we weren't relying on the tests to do anything for us in Jenkins. Historically, we ignored failures and proceeded to deploy regardless of the test outcome. 

> (Alternatively, could we just run tests in one environment during deploys instead?)

The way we currently think about this monolithic pipeline, there isn't an obvious solution to that. I _could_ see a path forward to making the tests optional, defaulting to no tests, and running these things with some set schedule. I'm not sure of the value proposition, though...

> Keeping our merge > deploy cycle short will minimize these risks [..]

Most definitely!